pkg/pipelines/environments/environments.go

```diff
@@ -64,6 +65,17 @@ func (b *envBuilder) Service(env *config.Environment, svc *config.Service) error
 	if _, ok := b.files[envBindingPath]; !ok {
 		b.files[envBindingPath] = createRoleBinding(env, envBasePath, b.cicdEnv.Name, b.saName)
 	}
+	// Add trigger bindings only if source url exists
+	if svc.SourceURL != "" {
+		repo, err := scm.NewRepository(svc.SourceURL)
+		if err != nil {
+			return err
+		}
+		prBinding, bindingName := repo.CreatePRBinding(b.cicdEnv.Name)
+		pipelinesPath := filepath.Join(envBasePath, "pipelines")
+		bindingPath := filepath.Join(pipelinesPath, "06-bindings", bindingName+".yaml")
+		b.files[bindingPath] = prBinding
+	}
 	return nil
```

Comment:
Is creating PRBinding being done in Init/Bootstrap?  It probably should be created only once.

Reply:
I've removed the bindings from the builder function. Now the bindings will only be created during  bootstrap/init  https://github.com/rhd-gitops-example/odo/blob/7d6b012a39ea9ceb8cd16c99d19af4799360bd8d/pkg/pipelines/init.go#L207